libr/bin/pdb/pdb_downloader.c

```diff
@@ -18,89 +18,77 @@ static bool checkExtract() {
 	return true;
 }
 
-static bool checkCurl() {
-	const char nul[] = R_SYS_DEVNULL;
-	if (r_sys_cmdf ("curl --version > %s", nul) != 0) {
+static bool download_and_write(SPDBDownloaderOpt *opt, const char *file) {
+	char *dir = r_str_newf ("%s%s%s%s%s",
+		opt->symbol_store_path, R_SYS_DIR,
+		opt->dbg_file, R_SYS_DIR,
+		opt->guid);
+	if (!r_sys_mkdirp (dir)) {
+		free (dir);
 		return false;
 	}
-	return true;
+	char *url = r_str_newf ("%s/%s/%s/%s", opt->symbol_server, opt->dbg_file, opt->guid, file);
+	int len;
+	char *file_buf = r_socket_http_get (url, NULL, &len);
+	free (url);
+	if (len && file_buf && *file_buf) {
```

Comment:
use `!R_STR_ISEMPTY` or `R_STR_ISNOTEMPTY`

Reply:
I'd make the error path nested, and leave the "main part" of the function outside the if, but that's just a personal taste actually :)

```suggestion
	if (!len || R_STR_ISEMPTY (file_buf)) {
		free (file_buf);
		return false;
	}
	char *path = ...
```